### ISSUE TITLE
Upgrading from polling to webtransport will report engine invalid WebTransport handshake.

### ISSUE THREAD

**Issue by zishang520** [socketio/engine.io#688]
**Describe the bug**
Upgrading from polling to webtransport will report engine invalid WebTransport handshake.

**To Reproduce**

Please fill the following code example:

Engine.IO server version: `6.5.1`

*Server*

```js
import { readFileSync } from "fs";
import { createServer } from "https";
import { Server } from "engine.io";
import { Http3Server } from "@fails-components/webtransport";

// WARNING: the total length of the validity period MUST NOT exceed two weeks (https://w3c.github.io/webtransport/#custom-certificate-requirements)
const cert = readFileSync("/path/to/my/cert.pem");
const key = readFileSync("/path/to/my/key.pem");

const httpsServer = createServer({
  key,
  cert
});

httpsServer.listen(3000);

const engine = new Server({
  transports: ["polling", "websocket", "webtransport"] // WebTransport is not enabled by default
});

engine.attach(httpsServer);

const h3Server = new Http3Server({
  port: 3000,
  host: "0.0.0.0",
  secret: "changeit",
  cert,
  privKey: key,
});

(async () => {
  const stream = await h3Server.sessionStream("/engine.io/");
  const sessionReader = stream.getReader();

  while (true) {
    const { done, value } = await sessionReader.read();
    if (done) {
      break;
    }
    engine.onWebTransportSession(value);
  }
})();

h3Server.startServer();
```

Engine.IO client version: `6.5.1`

*Client*

```js
const socket = require("engine.io-client")("wss://localhost:3000", {
    transports: ["polling", "webtransport"]
});

function send() {
    socket.send('ping');
}

socket.on('open', () => {
    send();
});

socket.on('close', () => {
    console.log("closed.")
});

socket.on('message', () => {
    // setTimeout(send, 100);
});
```

**Expected behavior**
```
  engine:socket executing batch send callback +4ms
  engine intercepting request for path "/engine.io/" +14ms
  engine handling "POST" http request "/engine.io/?EIO=4&transport=polling&t=Obthcpl&sid=tEe1aTFiB5e8y_T8AAAA" +0ms
  engine applying middleware n°1 +0ms
  engine setting new request for existing client +1ms
  engine:polling received "4ping" +14ms
  engine:socket received packet message +11ms
4
  engine:socket sending packet "message" (pong) +1ms
  engine intercepting request for path "/engine.io/" +7ms
  engine handling "GET" http request "/engine.io/?EIO=4&transport=polling&t=Obthcpm&sid=tEe1aTFiB5e8y_T8AAAA" +0ms
  engine applying middleware n°1 +0ms
  engine setting new request for existing client +1ms
  engine:polling setting request +7ms
  engine:socket flushing buffer to transport +6ms
  engine:polling writing "4pong" +0ms
  engine:socket executing batch send callback +0ms
handshake data >>>>>> 0{"sid":"tEe1aTFiB5e8y_T8AAAA"}2probe
  engine invalid WebTransport handshake +7ms
  engine intercepting request for path "/engine.io/" +31ms
  engine handling "GET" http request "/engine.io/?EIO=4&transport=polling&t=Obthcq1&sid=tEe1aTFiB5e8y_T8AAAA" +0ms
  engine applying middleware n°1 +0ms
  engine setting new request for existing client +1ms
  engine:polling setting request +39ms
```
```nodejs
        console.log(handshake);
        const sid = parseSessionId(handshake);
        if (!sid) {
            debug("invalid WebTransport handshake");
            return session.close();
        }
```

**Platform:**
 - Device: [e.g. Samsung S8]
 - OS: [e.g. Android 9.2]

**Additional context**
Upgrading from polling to webtransport will report engine invalid WebTransport handshake, because the original message reported is: 0{"sid":"xxx"}2probe, [parseSessionId](https://github.com/socketio/engine.io/blob/7dd1350bf93542282eba96ed6f5162dc249a7e65/lib/server.ts#L152) cannot correctly parse the sid

**Comment by darrachequesne** [comment]
That's weird, it seems two messages are concatenated: `0{"sid":"tEe1aTFiB5e8y_T8AAAA"}2probe`

Which version of Firefox are you using? Which OS?

**Comment by zishang520** [comment]
> That's weird, it seems two messages are concatenated: `0{"sid":"tEe1aTFiB5e8y_T8AAAA"}2probe`
> 
> Which version of Firefox are you using? Which OS?

The OS is the latest version of Windows 11, and the Firefox browser is also the latest version.

**Comment by darrachequesne** [comment]
@zishang520 this should be fixed by https://github.com/socketio/engine.io/commit/a306db09e8ddb367c7d62f45fec920f979580b7c, included in version `6.5.2`.

Could you please check?

**Comment by zishang520** [comment]
ok i will get back to you after testing.

**Comment by zishang520** [comment]
> @zishang520 this should be fixed by [a306db0](https://github.com/socketio/engine.io/commit/a306db09e8ddb367c7d62f45fec920f979580b7c), included in version `6.5.2`.
> 
> Could you please check?


I have tested sending and receiving messages, and there are no issues; it functions properly. However, there is another problem that needs to be addressed. Here are the reproduction steps:

On a Windows system, using the Firefox browser (version 115.0.2 (64-bit)), after starting the engine.io server, connecting to the server using the browser client, sending a message to the server in the 'open' event. Then, pressing F5 to refresh the page and reconnect to the server leads to a server crash.

```
  send accept ranges +0ms
  send cache-control public, max-age=0 +0ms
  send modified Fri, 04 Aug 2023 06:56:42 GMT +0ms
  send etag W/"5f3-189bf558e3d" +0ms
  send content-type application/javascript +0ms
  send not modified +0ms
  engine intercepting request for path "/engine.io/" +10m
  engine handling "GET" http request "/engine.io/?EIO=4&transport=polling&t=Oc_Oe50" +0ms
  engine applying middleware n°1 +0ms
  engine handshaking client "MKuE7QAV04RAXQT5AAAB" +1ms
  engine:transport readyState updated from undefined to open (polling) +10m
  engine:socket readyState updated from undefined to opening +3s
  engine:socket readyState updated from opening to open +1ms
  engine:socket sending packet "open" ({"sid":"MKuE7QAV04RAXQT5AAAB","upgrades":["webtransport"],"pingInterval":25000,"pingTimeout":20000,"maxPayload":1000000}) +0ms
  engine:polling setting request +10m
  engine:socket flushing buffer to transport +0ms
  engine:polling writing "0{"sid":"MKuE7QAV04RAXQT5AAAB","upgrades":["webtransport"],"pingInterval":25000,"pingTimeout":20000,"maxPayload":1000000}" +0ms
  engine:socket executing batch send callback +1ms
  engine:transport readyState updated from open to closed (webtransport) +16ms
  engine:socket readyState updated from open to closed +15ms
  engine:webtransport error while reading: undefined +3s
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "0".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v18.17.0
```

**Comment by darrachequesne** [comment]
@zishang520 this should finally be fixed by https://github.com/socketio/engine.io/commit/ff1c8615483bab25acc9cf04fb40339b0bd78812, included in version `6.5.3`.

Could you please confirm?

**Comment by zishang520** [comment]
> @zishang520 this should finally be fixed by [ff1c861](https://github.com/socketio/engine.io/commit/ff1c8615483bab25acc9cf04fb40339b0bd78812), included in version `6.5.3`.
> 
> Could you please confirm?

I have completed the test, and the issue has been fixed. Thank you.
